DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/SwiftUI/Image+Drawing.swift

```diff
@@ -0,0 +1,128 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+
+import CoreGraphics
+import Foundation
+import UIKit
+
+@available(iOS 13.0, tvOS 13.0, *)
+extension GraphicsImage {
+    init?(rasterizing drawingContents: NSObject, origin: CGPoint) {
+        let rasterizationScale = UIScreen.main.scale
+
+        guard let cgImage = Self.drawingRasterizer.image(
+            rasterizing: drawingContents,
+            origin: origin,
+            scale: rasterizationScale
+        ) else {
+            return nil
+        }
+
+        self.init(contents: .cgImage(cgImage), scale: rasterizationScale, orientation: .up)
+    }
+}
+
+@available(iOS 13.0, tvOS 13.0, *)
+extension GraphicsImage {
+    fileprivate class DrawingRasterizer: @unchecked Sendable {
+        private let cache = NSCache<NSNumber, CGImage>()
+        private let lock = NSRecursiveLock()
```

Comment:
question/ Do we need a lock? I guess this will always execute on the processing queue.